### PR TITLE
Added protoc-gen-doc plugin to generate API references

### PR DESCRIPTION
**Pull request by yfei1**
This commit added necessary Makefile commands to install `protoc-gen-doc` plugin to autogen markdown files for our APIs.

Simply run `make api/api.md`, then switch to open-match-docs repo, you will be able to see the auto-generated file lying under `site/content/en/docs/Reference/api.md`.

### CODE REVIEW

**Comment by yfei1** [comment]
@cla-bot rescan

**Comment by yfei1** [comment]
@googlebot rescan

File: Makefile [hunk 1]
```diff
@@ -669,6 +673,21 @@ api/%.swagger.json: api/%.proto third_party/ build/toolchain/bin/protoc$(EXE_EXT
 		-I $(REPOSITORY_ROOT) -I $(PROTOC_INCLUDES) \
 		--swagger_out=logtostderr=true,allow_delete_body=true:$(REPOSITORY_ROOT)
 
+api/api.md: third_party/ build/toolchain/bin/protoc-gen-doc$(EXE_EXTENSION)
```
Comment (sawagh) [inline Makefile]: lets go over the output tomorrow and see if the the autogen output is good enough - else we could use this tool one time to generate a starting point and then keep it updated manually.

Comment (yfei1) [reply Makefile]: The output has a PR in the open-match-docs repo

Comment (sawagh) [reply Makefile]: Looks pretty good. Looks like agones uses this too? Lets bake this in to our steps.

**Pull request merged**